### DISCUSSION TITLE
validateReceiptAndroid always "Error"

### DISCUSSION THREAD

**Discussion by akaoppa** [hyochan/react-native-iap#2024] [Q&A] [answered]
Hello,
I'm using one of the latest version of IAP 11.x.x something. But I always get  "Error"  that throw the catch from          RNIap.validateReceiptAndroid(purchase.packageNameAndroid, purchase.productId, purchase.purchaseToken, androidToken, false)

The android token from my backend server is ok. I successfully get a json when I'm trying to get it from URL :    "https://androidpublisher.googleapis.com/androidpublisher/v3/applications/"+purchase.packageNameAndroid+"/purchases/products/"+purchase.productId+"/tokens/"+purchase.purchaseToken+"/?access_token="+token;

So I tried to use it without validateReceiptAndroid but then I get : 'purchase is not suitable to be purchased' from finishTransaction(purchase, true);

I don't know what is it wrong ? I'm using a service account to access to get a token from my backend server and it's working fine. 
My service account is linked to the app in googleplay console, and the app is in closed alpha.
Old version of IAP 6.0.2 in my other project worked very well. So I don't know what is it wrong ?

Thank you!

**Comment by andresesfm** [comment]
@akaoppa Please post an example of your `purchase` object here: https://github.com/dooboolab/react-native-iap/issues/2021 seems to be a bug. Thank you

**Reply by akaoppa** [reply]
![8470B4F9-799F-4F32-ABE8-DBF91B213483](https://user-images.githubusercontent.com/60573069/194385984-cfdf5299-725c-4934-aee3-a22ff2dfe82e.jpeg)

im getting a normal purchase object but validateReceiptAndroid is not working it’s going to the catch directly as you can see in the screenshot

**Comment by andresesfm** [comment]
@akaoppa

1) What's the "Error" you get when you call `validateReceiptAndroid`
2) When you call that URL are you using a GET or a POST?
3) Finish transaction fails with that message: "purchase is not suitable to be purchased" when `purchase.purchaseToken` is not present. Can you please confirm that `purchaseToken` is in the `purchase` object when you call `finishTransaction`?

**Comment by akaoppa** [comment]
1. Error contain just "Error” and nothing else
2. Im using "await fetch"
3. Yes the purchasetoken is the same but im getting this error only im using the URL method

**Comment by andresesfm** [comment, marked as answer]
released fix on `12.0.3`

**Reply by akaoppa** [reply]
I just updated to 12.0.3 and still not working for me. validateReceiptAndroid always throw "Error"
and with direct fetch URL, the finishTransaction function throw "purchase is not suitable to be purchased"
I tried old version of the IAP 6.0.2 and this one work. Is there any specific requirement with recent version ? like minimum specific targetSdkVersion or something else to have? Because I have targetSdkVersion = 31 

![image](https://user-images.githubusercontent.com/60573069/194728778-2b6a88ae-5b88-4eac-9056-98ffb958ff25.png)

![image](https://user-images.githubusercontent.com/60573069/194728628-34e6bf44-8798-46bc-b7c4-6bdf0c8be44d.png)

purchase object contains, purchaseToken :

![image](https://user-images.githubusercontent.com/60573069/194729307-eb553201-a98a-4bec-aa0e-467ad09d4705.png)

**Reply by akaoppa** [reply]
I just saw that I have old version of react-native below 0.65.1, I'm going to update to 0.69.0 maybe that was the problem

**Reply by andresesfm** [reply]
I noted that in the first snippet `purchaseState` is 0 which means the purchase didn't go through

**Reply by akaoppa** [reply]
I managed to solve the problem thanks to your last commit.
The [documentation ](https://react-native-iap.dooboolab.com/docs/api-reference/methods/android/validate-receipt-android/
) was misleading, it's said to use :  validateReceiptAndroid(
      'your-package-name',
      'your-product-id',
      'your-product-token',
      'your-access-token',
      true,
    );

Which is missing the bracket "{ }" between

instead of using :             RNIap.validateReceiptAndroid({packageName: purchase.packageNameAndroid, productId: purchase.productId, productToken: purchase.purchaseToken, accessToken: androidToken, isSub: false})

**Reply by andresesfm** [reply]
Thank you for the feedback @akaoppa . I have fixed the documentation here: https://github.com/dooboolab/react-native-iap/pull/2041
